src/main/java/net/snowflake/client/core/SFSession.java

```diff
@@ -134,7 +133,7 @@ public boolean isSafeToClose() {
    * @return enum of type QueryStatus indicating the query's status
    * @throws SQLException
    */
-  public QueryStatus getQueryStatus(String queryID) throws SQLException {
+  public QueryStatus getQueryStatus(String queryID) throws SQLException, SFException {
```

Comment:
why not wrap `SFException` into `SQLException` so you don't need to expose the `SFException` here?

Reply:
SFException gets hidden again for the final getStatus() function. Basically, on line 173 in this function **catch (SnowflakeReauthenticationRequest | SFException ex)**, it was difficult to catch both types of exceptions and throw them both as SnowflakeSQLExceptions, depending on which was which. So I let the SFException be exposed temporarily but caught again later.

Reply:
This is true. Updated to SFException is now hidden inside SQLException instead.